### DISCUSSION TITLE
Is Rockford/Rocky still in charge of CSLA?

### DISCUSSION THREAD

**Discussion by LeeMcGraw** [MarimerLLC/csla#4257] [Questions]
Is Rockford/Rocky still in charge of CSLA?   I know him from the first and old days, and was wondering.  Please advise.  Lee McGraw, Enterprise Technical Architect.

**Comment by StefanOssendorf** [comment]
Yes he is.

**Reply by rockfordlhotka** [reply]
Hi Lee, yes, I am still very involved in CSLA, and am generally "in charge" - though there are some other very active contributors that are making a major difference in the codebase and capabilities of the framework.

**Reply by JasonBock** [reply]
Oh, c'mon @rockfordlhotka, you're the boss here 😆

**Reply by LeeMcGraw** [reply]
@rockfordlhotka 

HEY buddy.  Long time no convo.  I hope you're doing well.  I figured you were a millionaire retired on some island by now, getting your feet rubbed by some ladies. Gotta be getting sick of that Minnesota weather.  Anyway, just wanted to look what's new with your library. 

Btw, do you have a dependency injection module in the library?  I recently wrote one (took me a day), in dotnetfiddle, and could send you the link.  It's pretty straight forward, but still is better than MS's DI extension (not hard to accomplish).

Anyway, good to hear from you and I'm going to go back to reading this updated codebase.  Let me know about the DI.  U have my email address I think.

R. Lee McGraw

**Reply by LeeMcGraw** [reply]
![image](https://github.com/user-attachments/assets/81ff5ee9-a46c-4e05-88c9-66623f63ab30)

**Reply by rockfordlhotka** [reply]
> Oh, c'mon @rockfordlhotka, you're the boss here 😆

I try to be a benevolent boss though 😉

**Reply by rockfordlhotka** [reply]
@LeeMcGraw that goes _way_ back doesn't it?

**Reply by rockfordlhotka** [reply]
> @rockfordlhotka
> 
> Btw, do you have a dependency injection module in the library? I recently wrote one (took me a day), in dotnetfiddle, and could send you the link. It's pretty straight forward, but still is better than MS's DI extension (not hard to accomplish).

CSLA relies on the DI model built into .NET - `ServiceProvider`, `IServiceCollection`, etc.

There used to be (and maybe still are?) some DI alternatives, but if they aren't compatible with the .NET interfaces then you can't use them with aspnetcore or Blazor - so they aren't very useful in real life.

**Comment by StefanOssendorf** [comment]
I changed it to a discussion because it's not really an issue :)

**Comment by Chicagoan2016** [comment]
@LeeMcGraw , Out of sheer curiosity, when did you last work with Csla?

**Reply by LeeMcGraw** [reply]
It's been a looooong time since I even wrote code as a job title. I am an Enterprise Architect now, but I worked with CSLA before it was called CSLA, back when Rocky working his Wrox Business Objects in VB5 book.   I guess I had some interaction with CSLA about 12 yrs ago on a project, but I wasnt in charge of that project.  I still, however, look at the old code from time to time to remember how he did something.  

Why do you ask?

**Reply by LeeMcGraw** [reply]
Dont get me wrong. I write code ALOT still, but mostly for fun, not for job.
